xplique/metrics/fidelity.py

```diff
@@ -383,16 +408,11 @@ def __init__(self,
 
         assert 0 < max_percentage_perturbed <= 1, \
             "max_percentage_perturbed should be between 0 and 1"
-        max_nb_perturbation = floor(max_percentage_perturbed*self.nb_features)
+        self.max_percentage_perturbed = max_percentage_perturbed
 
         if steps == -1:
-            steps = max_nb_perturbation
-        assert 1 < steps <= max_nb_perturbation, \
-            f"The number of steps must be between 2 and {max_nb_perturbation} (or '-1')."
+            steps = self.max_percentage_perturbed * self.nb_features
```

Comment:
I think steps should be an int. Thus there is a floor or ceil function missing I guess

Reply:
Why don't you keep max_nb_pertubation ?

Reply:
The idea was to limit the number of attribute of the class, we discussed it with Thomas, do you think it is not justified ?